room/compiler-processing/src/main/java/androidx/room/compiler/processing/XAnnotation.kt

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2021 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.room.compiler.processing
+
+/**
+ * This wraps annotations that may be declared in sources, and thus not representable with a
+ * compiled type. This is an equivalent to the Java AnnotationMirror API.
+ *
+ * Values in the annotation can be accessed via [valueArguments], the [XAnnotation.get] extension
+ * function, or any of the "getAs*" helper functions.
+ *
+ * In comparison, [XAnnotationBox] is used in situations where the annotation class is already
+ * compiled and can be referenced. This can be converted with [asAnnotationBox] if the annotation
+ * class is already compiled.
+ */
+interface XAnnotation {
+    /**
+     * The simple name of the annotation class.
+     */
+    val simpleName: String
```

Comment:
might want to name this `name` because all other elements use `name`.

Reply:
do we also need qualifiedName ?

Reply:
qualifiedName can be access from `xannotation.type.typeelement. qualifiedName`. I provided simple name because to enable access without resolving the type as an optimization.

I can also expose qualifiedName directly as a property to avoid the deep chain needed to otherwise access it if you think it would be a nice thing to have

Reply:
we can. btw, i'm not sure if you can do early filtering by name due to typealiases.